src/Servers/Kestrel/Core/src/Internal/Http/HttpProtocol.cs

```diff
@@ -917,7 +917,15 @@ public void ProduceContinue()
                 ((IHeaderDictionary)HttpRequestHeaders).TryGetValue(HeaderNames.Expect, out var expect) &&
                 (expect.FirstOrDefault() ?? "").Equals("100-continue", StringComparison.OrdinalIgnoreCase))
             {
-                Output.Write100ContinueAsync().GetAwaiter().GetResult();
+                ValueTask<FlushResult> vt = Output.Write100ContinueAsync();
```

Comment:
Some of these look like they could use our extension methods
https://github.com/dotnet/aspnetcore/blob/main/src/Shared/ValueTaskExtensions/ValueTaskExtensions.cs

Reply:
I don't think either of those apply here...?

Reply:
I believe the suggestion here is to change the original code to this:

```
Output.Write100ContinueAsync().GetAsTask().GetAwaiter().GetResult();
```

The `GetAsTask()` already avoids the task allocation for cases where the value task is already completed.

Reply:
Doing so would unnecessarily allocate a Task if this ValueTask completed as Canceled or Faulted.  But, otherwise, sure, it could do that.

Reply:
I agree with the interpretation, but I can't speak to the impact of that on this code.

Reply:
We need to fix this code to not be blocking 😄